### PR TITLE
feat(sdk): `defineEmbeddingSdkConfig` to make typing easier 

### PR DESCRIPTION
**Pull request by npretto**
tldr: do what vite (and other tools) do: https://vite.dev/config/#config-intellisense

In the past couple of weeks I had to edit some configuration for libraries/tools etc, and one of the thing that annoyed me the most was when the configs were in plain js as it wasn't typed. It's not a good "first time" experience.

For JS users we're kinda on the same boat, if they do
```
const config = {}

const Component = () => {
  return <MetabaseProvider config={config}>...</MetabaseProvider>
 }
```
They won't have any auto complete or docs available.

Similarly in typescript, they have to type it as `SDKConfig`, which they need to know even exists.

For this reason I propose we add this `defineEmbeddingSdkConfig` function *and use it on all code snippets/samples we have*.

The function is just to add the type, it literally does nothing once built, the benefits are:

- js users will have autocomplete and docs depending on their setup (for vscode this should work out of the box)
- the code snippets will be the same between js and ts, as no explicit type needs to be written in the them
- for ts users, the main benefit is that we can make all code samples use the function, and they'll have type safety at the config definition level without having to type it manually

## Demo
image

### CODE REVIEW

**Review by oisincoveney — APPROVED** [review]
this checks out

**Review by albertoperdomo — APPROVED** [review]

File: enterprise/frontend/src/embedding-sdk/README.md [hunk 1]
```diff
@@ -714,7 +714,8 @@ You can provide a theme object to the `MetabaseProvider` to customize the look a
 Here is the full list of theme properties supported. All of them are optional.
 
 ```ts
-const theme = {
+import {defineEmbeddingSdkTheme} from "@metabase/embedding-sdk-react";
+const theme = defineEmbeddingSdkTheme({
     // Specify a font to use from the set of fonts supported by Metabase.
     // You can set the font to "Custom" to use the custom font
```
Comment (oisincoveney) [inline enterprise/frontend/src/embedding-sdk/README.md:720]: Is this going to be weird for clients? Maybe we should even change the name from "custom" to something else (maybe "metabase instance font" and transform it on the FE when we're parsing the config

**Review by oisincoveney — APPROVED** [review]
Looks good, just one question

File: enterprise/frontend/src/embedding-sdk/README.md (hunk 1 above)
Comment (npretto) [reply enterprise/frontend/src/embedding-sdk/README.md:720]: I found it confusing too, but I think that's what they select in the UI on metabase too though.
It should pickup that by default anyway though if it's configured on their instance
